src/call/mod.rs

```diff
@@ -683,4 +686,44 @@ impl SinkBase {
         self.batch_f.take();
         Poll::Ready(Ok(()))
     }
+
+    #[inline]
+    fn poll_flush<C: ShareCallHolder>(
+        &mut self,
+        cx: &mut Context,
+        call: &mut C,
+    ) -> Poll<Result<()>> {
+        if self.batch_f.is_some() {
+            ready!(self.poll_ready(cx)?);
+        }
+        if self.buf_flags.is_some() {
+            // Set `buffer_hint` to false to indicate that this is the last message.
+            self.start_send_buffer_message(false, call)?;
```

Comment:
What if all messages are pushed into the sink with `buffer hint = false`?

Reply:
Fixed